### ISSUE TITLE
no error on unkown variable of interface

### ISSUE THREAD

**Issue by gitawego** [Microsoft/TypeScript#13836]
**TypeScript Version:**  2.1.5

**Code**

```ts
interface ProductionRecord {
  _id: string,
  productName: string;
  price: number;
  quantityProduced: number;
}

type ProductionRecords = ProductionRecord[];

interface ProductionRecordsState {
  records: ProductionRecords
}

const initialState: ProductionRecordsState = {
  records: [],
  test:"my test string"
};

```

**Expected behavior:**

vscode or tsc should give error about defining unknown attribute ``test``

**Actual behavior:**

no error at all when defining the ``initialState``. but it gives error when access the variable :

```ts
// it give error when use it, it says: can not find ``test`` in type ``ProductionRecordsState``
const test = initialState.test;
```

**Comment by hxuanhung** [comment]
same issue

**Comment by vasa-c** [comment]
But in fact the entity that implements the interface must contain all of its properties. But it not disallowed to have others.

**Comment by gitawego** [comment]
@vasa-c it's not true, otherwise, when accessing the variable ``test``, vscode or tsc wouldn't show errors.

in your case, interface ``ProductionRecordsState`` must be defined as:

```ts
interface ProductionRecordsState {
  records: ProductionRecords
  [key:string]:any;
}
```

**Comment by vasa-c** [comment]
> in your case, interface ProductionRecordsState must be defined as:

It solves second problem, `initialState.test` will not throw error. :)

But

```ts
{
  records: [],
  test:"my test string"
};
```

It object implements `ProductionRecordsState` and in additional has property `test`

**Comment by gitawego** [comment]
anyway, the old behaviour, and I think it was the correct one, is you must define exactly the same attributes of an interface. if not. what's the meaning of implementing ``[key:string]:any`` ?

if interface ``ProductionRecordsState `` is:
```ts
interface ProductionRecordsState {
  records: ProductionRecords;
}
```
this must throw error
```ts
const initialState: ProductionRecordsState = {
  records: [],
  test:"my test string"
};
```

the only way of preventing error is by defining like this:
```ts
interface ProductionRecordsState {
  records: ProductionRecords;
  [key:string]:any;
}
```

**Comment by hxuanhung** [comment]
@gitawego tested on [http://www.typescriptlang.org/play/index.html](http://www.typescriptlang.org/play/index.html) and it throws error as expected.

**Comment by gitawego** [comment]
@hxuanhung don't know which typescript version is in the playground.
